aggregator/aggregator.go

```diff
@@ -192,14 +221,30 @@ func (agg *aggregator) Open() error {
 	defer agg.Unlock()
 
 	if agg.state != aggregatorNotOpen {
-		return errAggregatorIsOpenOrClosed
+		return errAggregatorAlreadyOpenOrClosed
```

Comment:
For sure, maybe calling it aggregatorUninitialized would be better? Bit confusing between NotOpen and Closed.